File: src/p0260/java/solution.java
class Solution {
    public int[] singleNumber(int[] nums) {
        int d = 0, x = 0, y = 0;
        for (int e : nums) {
            d ^= e;
        }
        d = d & (-d);
        for (int e : nums) {
            if ((d & e) == 0) {
                x ^= e;
            } else {
                y ^= e;
            }
        }
        return new int[]{x, y};
    }
}

File: src/p0260/cpp/solution.cpp
#include <iostream>
#include <vector>

using namespace std;


class Solution {
public:
    vector<int> singleNumber(vector<int> &nums) {
        int d = 0, x = 0, y = 0;
        for (auto e : nums) {
            d ^= e;
        }
        d = d & (-d);
        for (auto e : nums) {
            (d & e) ? (x ^= e) : (y ^= e);
        }
        return {x, y};
    }
};
